src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -468,6 +468,8 @@ protected void CreateSymbolicLink_PathToTarget_RelativeToLinkPath_Internal(bool
             // Verify that Target is resolved and is relative to Link's directory and not to the cwd.
             Assert.False(targetInfo.Exists);
             Assert.Equal(Path.GetDirectoryName(linkInfo.FullName), Path.GetDirectoryName(targetInfo.FullName));
+
+            Directory.SetCurrentDirectory(Path.GetTempPath());
```

Comment:
> The fix was to set the current directory back at the end of the remote-exec'd code.

you mean set it back to the path we previously set in `ChangeCurrentDirectory()`? If so, I think this should be `Directory.SetCurrentDirectory(tempCwd)`

Reply:
We don't want that -- tempCwd is within the test's folder, which we want cleaned up.

(As I mentioned in the description, I don't understand why the current directory of the child process is relevant since by the time the test exits, the remoteexec process has quit and presumably the OS closed any handles it had open. But nevertheless that is what I saw.)

Reply:
Ah I see what you mean, sounds like the folder is still locked by the RemoteExecutor process. 
I don't know if it may be possible to change the CWD of the RemoteExecutor Process in, say, `RemoteInvokeHandle.Dispose()` in order to avoid doing this each time its cwd gets changed.

Reply:
That may be cleaner.

@jkotas do you have any idea why resetting the current directory seems necessary, given the remote exec process exits before we clean up the test folder?

Reply:
created https://github.com/dotnet/arcade/pull/9200

Reply:
> @jkotas do you have any idea why resetting the current directory seems necessary, given the remote exec process exits before we clean up the test folder?

My guess is that it is likely anti-virus, watson or some other kind of Windows background task holding the directory. I think that it is unlikely for the process itself to be holding onto the directory once it exists.